### DISCUSSION TITLE
Namespace not found

### DISCUSSION THREAD

**Discussion by UltraWelfare** [spatie/laravel-data#887] [Q&A]
After creating some data classes and using `php artisan typescript:transform` to make the TS types, I'm unable to use them.

The generated file is as follows:
```ts
declare namespace App.Actions.Customers {
export type CreateCustomerData = {
name: string;
email: string;
phone: string | null;
};
export type CustomerData = {
id: number;
name: string;
email: string;
phone: string | null;
};
}
```

When trying to use this namespace `App.Actions.Customer` typescript errors that the namespace cannot be found.
Exporting the declared namespace fixes the problem, I wonder why though the namespace isn't exported from the transformer in the first place?

It feels wrong to touch generated code and tedious to do it each time you make changes

**Comment by UltraWelfare** [comment]
Found the issue quite quickly :P

The tsconfig searches for the `.d.ts` files on the `resources/js` folders.
 `"include": ["resources/js/**/*.ts", "resources/js/**/*.tsx", "resources/js/**/*.d.ts"]`

However the command generates the file in `resources/types/generated.d.ts`. I just had to change the output file in the `typescript-transformer.php` config:

`output_file' => resource_path('js/types/generated.d.ts'),`

and it's all working right now.

I used laravel installer with breeze and react (Inertia). Would be nice to include it somewhere in the docs:)

**Reply by christopherokonkwo** [reply]
Thanks. This saved me a lot of googling time.
